src/main/java/io/jenkins/plugins/opentelemetry/job/JunitTaskHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The Original Author or Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.jenkins.plugins.opentelemetry.job;
+
+import hudson.Extension;
+import hudson.tasks.junit.pipeline.JUnitResultsStep;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.Tracer;
+import jenkins.YesNoMaybe;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepAtomNode;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Customization of spans for {@code junit} step.
+ */
+@Extension(optional = true, dynamicLoadable = YesNoMaybe.YES)
+public class JunitTaskHandler implements StepHandler {
+    @Override
+    public boolean canCreateSpanBuilder(@Nonnull FlowNode flowNode, @Nonnull WorkflowRun run) {
+        return flowNode instanceof StepAtomNode && ((StepAtomNode) flowNode).getDescriptor() instanceof JUnitResultsStep.DescriptorImpl;
+    }
+
+    @Nonnull
+    @Override
+    public SpanBuilder createSpanBuilder(@Nonnull FlowNode node, @Nonnull  WorkflowRun run, @Nonnull Tracer tracer) throws Exception {
```

Comment:
I introduced a method `StepHandler#afterSpanCreated()` that seems to be a good match for this use case.